src/main/java/LinkedStack.java

```diff
@@ -73,19 +79,19 @@ public boolean equals(Object o) {
             return false;
         }
         LinkedStack<?> that = (LinkedStack<?>) o;
-        if (this.size == that.size) {
-            Node<?> thisCurrent = this.top;
-            Node<?> thatCurrent = that.top;
-            while (thisCurrent != null) {
-                if (!thisCurrent.data.equals(thatCurrent.data)) {
-                    return false;
-                }
-                thisCurrent = thisCurrent.getNext();
-                thatCurrent = thatCurrent.getNext();
+        if (this.size() != that.size()) {
+            return false;
+        }
+        Node<?> thisCurrent = this.top;
+        Node<?> thatCurrent = that.top;
+        while (thisCurrent != null) {
+            if (!Objects.equals(thisCurrent.getData(), thatCurrent.getData())) {
```

Comment:
Reading this, it seems like `thatCurrent` can be `null`. May be worth a comment indicating that it can't be `null` since the sizes of the stacks are checked to be equal first.

Reply:
I added a comment, but I'll actually just change it to be `while (thisCurrent != null && thatCurrent != null)` for the same of understandableness and defensiveness.